app/controllers/followships_controller.rb

```diff
@@ -0,0 +1,52 @@
+class FollowshipsController < ApplicationController
+  def create
+    @followship= current_user.followships.build(followee_id: params[:followee_id],accepted: false)
+    if @followship.save
+      respond_to do |format|
+        format.js
+      end
+
+      Notification.notifications_creator(followship:@followship,params:params,type:"reverse_followship")
+
+    else
+      redirect_to users_path
+    end
+  end
+
+  def destroy
+    @followship= Followship.find_by_id(params[:id])
+    if @followship.destroy
+      respond_to do |format|
+        format.js
+      end
+      if @followship.user_id != current_user.id
+        Notification.notifications_creator(followship:@followship,params:params,type:"followship")
+
+
+      elsif @followship.user_id === current_user.id && @followship.accepted
+        Notification.notifications_creator(followship:@followship,params:params,type:"reverse_followship")
+
```

Comment:
This is not place to call the notification logic. Move the above logic to Notification and create the necessary method there. Declare Types in appropriate class.